### PR TITLE
Add AWS m6i type machines

### PR DESCRIPTION
**Pull request by patrickhuy**
**What this PR does / why we need it**: 
This PR adds new AWS m6i type machinesto ec2_instance_types.go in reference to the landscape setup

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Added support for m6i type AWS machine types
```

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @patrickhuy for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: cluster-autoscaler/cloudprovider/aws/ec2_instance_types.go [hunk 1]
```diff
@@ -1516,6 +1516,66 @@ var InstanceTypes = map[string]*InstanceType{
...
+		MemoryMb:     262144,
+		GPU:          0,
+	},
+	"m6i.24xlarge": {
+		InstanceType: "m6i.24xlarge",
+		VCPU:         96,
+		MemoryMb:     393216,
+		GPU:          0,
+	},
+  "m6i.32xlarge": {
```
Comment (prashanth26) [inline cluster-autoscaler/cloudprovider/aws/ec2_instance_types.go]: Alignment issue? Please fix it and make sure all attributes are aligned.

Comment (patrickhuy) [reply cluster-autoscaler/cloudprovider/aws/ec2_instance_types.go]: thanks! I corrected it!

**Review by prashanth26 — APPROVED** [review]
/lgtm

/invite @AxiomSamarth

**Review by AxiomSamarth — APPROVED** [review]
/lgtm

**Pull request merged**